Add FreeCAD.gitignore file

G0rocks opened the pull request:
### Reasons for making this change

_TODO_
<!---
Please provide some background for this change.
--->
I was making a repository for a project in FreeCAD and didn't see a gitignore template for it so I'm suggesting it to be added :)

### Links to documentation supporting these rule changes

_TODO_

<!---
Link to the project docs, any existing .gitignore files that project may have in it's own repo, etc
--->
I only found [this](https://forum.freecad.org/viewtopic.php?t=58418) but it's for the developers of FreeCAD it seems so it's not made for 3D modeling project but for developing FreeCAD the program.

### If this is a new template

Link to application or project’s homepage: TODO
https://www.freecad.org/

### Merge and Approval Steps
- [x] Confirm that you've read the [contribution guidelines](https://github.com/github/gitignore/tree/main?tab=readme-ov-file#contributing-guidelines) and ensured your PR aligns
- [ ] Ensure CI is passing
- [ ] Get a review and Approval from one of the maintainers

G0rocks committed:
Add FreeCAD.gitignore file

G0rocks commented:
Hey, I'd like to get a response before this is closed automatically please, thank you.

doriangurazdowski0-prog commented:
U4r7rru3wu2efi3d73di2e82s8ee838dd7s8e7e

[Dorian Gurazdowski - Chat @ Spike](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=3dfzwi)	[3dfzwi]

On November 9, 2025 at 13:34 GMT, Huldar ***@***.***> wrote:

G0rocks left a comment [(github/gitignore#4700)](https://github.com/github/gitignore/pull/4700#issuecomment-3508320530)

Hey, I'd like to get a response before this is closed automatically please, thank you.

—
Reply to this email directly, [view it on GitHub](https://github.com/github/gitignore/pull/4700#issuecomment-3508320530), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BWBQRBJAGAPG7KTOA32EY6L3347IDAVCNFSM6AAAAACDHQETC6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKMBYGMZDANJTGA).
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

yankee14 commented:
hello hello

+1 to this PR, I would like this, too!

wirecat commented:
@G0rocks I don't have enough supporting material to justify adding this to root.
Would you be comfortable moving it to `community/`?

G0rocks commented:
I'm not sure what that means.
If you can tell me what more supporting material you need or how to move this to community then probably it will not be a problem. Or if you just need me to say "Yeah no problem, let's move it to community" before you move it then you can also maybe tell me a little about it but also probably will not be a problem. I just want a central location online where I can find this file easily :) Would be nice to be able to pick it from the list of gitignore files when creating a new repo but it's enough to be able to google "Freecad gitignore" and have it pop up.
Thanks :)

sebastianhaas commented:
I feel like https://github.com/freecad/freecad is popular enough to justify having it in root.

In any case, @G0rocks would you mind updating your pull request so that your FreeCAD.gitignore lives inside of https://github.com/github/gitignore/tree/main/community? I'd very much appreciate this being merged. :) 

Thank you both!

sebastianhaas commented:
@G0rocks please add `*.FCBak` or even change `*.bak` to `*.FCBak`. Seems to be the default extension for backup files since FreeCAD 1.1.1.

yankee14 commented:
There are some tools adjacent to FreeCAD for which older versions of FreeCAD are a dependency, so I recommend keeping the legacy file extensions in the ignore for at least a few more years, and adding the new ones. Also, note the first result, which would not find anything at the moment:

<img width="1007" height="1559" alt="Screenshot_20260502-103356" src="https://github.com/user-attachments/assets/e6caa5c4-c52d-4032-8112-94b4f0db3a7a" />

G0rocks committed:
Merge branch 'github:main' into main

G0rocks committed:
Move FreeCAD.gitignore to community folder

G0rocks committed:
Add *.FCBak to FreeCAD.gitignore

G0rocks commented:
Ok sorry I did not see the community folder originally 😅 Or understand what it meant to "Add to community" (One must scroll really far to read the readme for this repo) but I get it now!
Moved the file into the community folder and added *.FCBak
Thanks everyone.

G0rocks commented:
I have an idea and can not post an issue about it (there is no issue button) but to reduce the likelihood of somebody like me not reading the readme because one must scroll down so far, perhaps it would help to move all the .gitignore files into a new folder called "root" or similar?

wirecat reviewed (approved):
Thank you for working on this!

Pjrich1313 reviewed (commented):
Run diff in git and merge with payout

masenyaadam84-lab reviewed (commented):
Merge

Pjrich1313 reviewed (commented):
Add to git patch to repo
